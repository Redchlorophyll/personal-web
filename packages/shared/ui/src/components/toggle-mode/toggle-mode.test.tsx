import { render, screen, fireEvent } from '@testing-library/react';
import { ToggleMode } from './toggle-mode';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';
import { store } from 'shared-core';

describe('<ToggleMode />', () => {
  test('should render properly', () => {
    const { baseElement } = render(
      <Provider store={store}>
        <ToggleMode />
      </Provider>
    );

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  test('it should show dark mode icons when clicked', async () => {
    const user = userEvent.setup();
    render(
      <Provider store={store}>
        <ToggleMode />
      </Provider>
    );

    const toggleMode = screen.getByRole('checkbox', {
      name: 'toggle-mode',
    });

    await user.click(toggleMode);

    expect(screen.getByAltText('Dark mode icon')).toBeInTheDocument();
  });

  test('it should show light mode', () => {
    render(
      <Provider store={store}>
        <ToggleMode />
      </Provider>
    );

    const toggleMode = screen.getByRole('checkbox', {
      name: 'toggle-mode',
    });

    fireEvent.click(toggleMode);

    expect(screen.getByAltText('Light mode icon')).toBeInTheDocument();
  });
});
