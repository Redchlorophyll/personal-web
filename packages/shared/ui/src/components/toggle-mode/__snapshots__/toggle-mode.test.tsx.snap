// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleMode /> should render properly 1`] = `
<body>
  <div>
    <div
      class="bg-black-300 w-24 h-12 rounded-full relative left transition-all duration-700"
    >
      <input
        aria-label="toggle-mode"
        class="opacity-0 peer w-24 h-24 z-10"
        id="toggle-mode"
        type="checkbox"
      />
      <label
        class="rounded-full absolute bg-black-100 p-1 left-2 top-1 peer-checked:bg-black-800 peer-checked:left-12 transition-all duration-1000 w-10 h-10 justify-center"
        for="toggle-mode"
      >
        <svg
          class="scale-90 -translate-x-[2px]"
          data-testid="light-mode-icon"
          fill="none"
          height="29"
          width="35"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#FFF494"
            height="21.757"
            rx="10.879"
            width="21.757"
            x="0.689"
            y="0.143"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="10.678"
            y="6.848"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="10.678"
            y="6.848"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="10.678"
            y="6.848"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="16.562"
            y="11.09"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="16.562"
            y="11.09"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="16.562"
            y="11.09"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="6.3"
            y="11.637"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="6.3"
            y="11.637"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="6.3"
            y="11.637"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="1.51"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="1.51"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="1.51"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="9.173"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="9.173"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="9.173"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="15.604"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="15.604"
            y="15.743"
          />
          <rect
            fill="#D2FFFF"
            height="12.316"
            rx="5.884"
            width="11.768"
            x="15.604"
            y="15.743"
          />
          <path
            d="M22.447 21.763a5.884 5.884 0 0 1 11.768 0v.548a5.884 5.884 0 1 1-11.769 0v-.548Z"
            fill="#D2FFFF"
          />
          <path
            d="M22.447 21.763a5.884 5.884 0 0 1 11.768 0v.548a5.884 5.884 0 1 1-11.769 0v-.548Z"
            fill="#D2FFFF"
          />
          <path
            d="M22.447 21.763a5.884 5.884 0 0 1 11.768 0v.548a5.884 5.884 0 1 1-11.769 0v-.548Z"
            fill="#D2FFFF"
          />
        </svg>
      </label>
    </div>
  </div>
</body>
`;
